src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -99,20 +103,20 @@ class RedownloadService[F[_]: NonEmptyParallel](
       (updated, ())
     }
 
-  def persistPeerProposal(peer: Id, height: Long, hash: String, reputation: Reputation): F[Unit] =
+  def persistPeerProposal(peer: Id, height: Long, hash: String, reputation: Reputation, signature: HashSignature): F[Unit] =
     logger.debug(s"Persisting proposal of ${peer.short} at height $height and hash $hash") >> peersProposals.modify {
       m =>
         val existing = m.getOrElse(peer, Map.empty)
         val updated =
           if (existing.contains(height)) m
-          else m.updated(peer, existing + (height -> PersistedSnapshotProposal(hash, reputation)))
+          else m.updated(peer, existing + (height -> PersistedSnapshotProposal(hash, height, signature, reputation)))
         val ignored = InvertedMap(updated.mapValues(a => takeHighestUntilKey(a, getRemovalPoint(maxHeight(a)))))
         (ignored, ())
     }
 
   def persistCreatedSnapshot(height: Long, hash: String, reputation: Reputation): F[Unit] =
     createdSnapshots.modify { m =>
-      val updated = if (m.contains(height)) m else m.updated(height, PersistedSnapshotProposal(hash, reputation))
+      val updated = if (m.contains(height)) m else m.updated(height, PersistedSnapshotProposal(hash, height, SignHelp.hashSign(hash, keyPair), reputation))
       val max = maxHeight(updated)
       val removalPoint = getRemovalPoint(max)
       val limited = takeHighestUntilKey(updated, removalPoint)
```

Comment:
If you expand the code below, you will find a method `persistAcceptedSnapshot` which accept snapshots (not snapshot proposals) from other peers. Should it stay as it is, or should we change it? @kpudlik @marcinwadon